cmd/cli/main.go

```diff
@@ -0,0 +1,361 @@
+						Usage:       "Optional ; Set RestoreToS3SecretPasswordKey, key inside secret, under which Restic repo password is stored, via cli or via env: ",
+						EnvVars: []string{
+							"RESTORETOS3SECRETPASSWORDKEY",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Namespace,
+						Required:    false,
+						DefaultText: "default",
+						Name:        "namespace",
+						Usage:       "Optional ; Set namespace in which You want to execute restore, via cli or via env: ",
+						EnvVars: []string{
+							"NAMESPACE",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Kubeconfig,
+						Required:    false,
+						DefaultText: "~/.kube/config",
+						Name:        "kubeconfig",
+						Usage:       "Optional ; Set kubeconfig to connect to cluster, via cli or via env:",
+						EnvVars: []string{
+							"KUBECONFIG",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func RunRestore(ctx *cli.Context) error {
+	var restoreName, snapshot string
+	var rootUser int64
+	var s3, pvc v1.RestoreMethod
+	logger := cmd.AppLogger(ctx).WithName("cli-restore")
+	// check permissions for user, defaults to 0 if not set correctly
+	rootUser, err := strconv.ParseInt(restore.Cfg.RunAsUser, 10, 64)
+	if err != nil {
+		rootUser = 0
+	}
+	// avoid name crashes + generate name if naming is not important for user
+	if restore.Cfg.RestoreName == "" {
+		restoreName = "cli-restore-" + restore.RandomStringGenerator(5)
+		log.Println("Creating restore with name: ", restoreName, " in napespace: ", restore.Cfg.Namespace)
+	} else {
+		restoreName = restore.Cfg.RestoreName
+	}
+
+	kconfig, err := clientcmd.BuildConfigFromFlags("", restore.Cfg.Kubeconfig)
+	if err != nil {
+		logger.Error(err, "Failed to create KubeConfig")
+		return err
+	}
+	logger.Info("Found Kubeconfig", "KUBECONFIG", restore.Cfg.Kubeconfig)
+
+	client, err := kubernetes.NewForConfig(kconfig)
+	if err != nil {
+		logger.Error(err, "Failed to create Kubernetes Client")
+		return err
+	}
+
+	crd, err := client.RESTClient().Get().Namespace("default").AbsPath("/apis/k8up.io/v1/").Resource("snapshots").DoRaw(context.TODO())
+	if err != nil {
+		logger.Error(err, "Failed to query Kubernetes api for snapshots, is it correct cluster and namespace? Is k8up installed correctly?")
+		return err
+	}
+
+	snpList := &v1.SnapshotList{}
+
+	err = json.Unmarshal(crd, &snpList)
+	if err != nil {
+		logger.Error(err, "Failed to unmarshal Snapshots")
+		return err
+	}
+	// kubectl get snapshots returns 8 character values, while snapshots itself are much longer
+	// so it's simple enough to check prefix
+	for _, snap := range snpList.Items {
+		if strings.HasPrefix((*snap.Spec.ID), restore.Cfg.Snapshot) {
+			snapshot = (*snap.Spec.ID)
+			break
+		}
+	}
+	// nothing to do if there are no snapshots
+	if len(snapshot) == 0 {
+		logger.Error(err, "Snapshot ID wasn't found")
+		return err
+	}
+	logger.Info("Found correct snapshot", "SNAPSHOT", snapshot)
```

Comment:
Nit: maybe make this a debug message.
```suggestion
	logger.V(1).Info("Found correct snapshot", "SNAPSHOT", snapshot)
```